packages/dom/src/dom.js

```diff
@@ -134,9 +134,21 @@ export function isHorizontalEdge( container, isReverse ) {
 
 	while ( node !== container ) {
 		let next = node[ `${ order }Sibling` ];
+		let hadBR;
+
+		// Skip over empty text nodes and at most one BR element if not checking
+		// the reverse. One BR element at the end is ok as it is invisible.
+		// `<p>test<br></p>` looks the same as `<p>test</p>`.
```

Comment:
Note that we would have to fix this issue _regardless_ of whether or not we add the trailing and padding `<br>` element. When inserting a double line break at the end of line (two are needed to see a new line), and you continue to write, you also end up with a trailing line break (which will wrongly be saved). `isHorizontalEdge` also requires this fix in that case.